--- cyrus-imapd-2.4.17/lib/auth_unix.c.authid_normalize	2013-07-05 21:15:20.219577370 +0200
+++ cyrus-imapd-2.4.17/lib/auth_unix.c	2013-07-05 21:15:20.266577773 +0200
@@ -156,10 +156,12 @@ const char *identifier;
 size_t len;
 {
     static char retbuf[81];
+    char backup[81];
     struct group *grp;
     char sawalpha;
     char *p;
     int username_tolower = 0;
+    int ic,rbc;
 
     if(!len) len = strlen(identifier);
     if(len >= sizeof(retbuf)) return NULL;
@@ -211,6 +213,22 @@ size_t len;
     /* now we don't */
     /* if (!sawalpha) return NULL;  */
 
+    if( (libcyrus_config_getswitch(CYRUSOPT_NORMALIZEUID) == 1) ) {
+        strcpy(backup,retbuf);
+       /* remove leading blanks */
+       for(ic=0; isblank(backup[ic]); ic++);
+       for(rbc=0; backup[ic]; ic++) {
+            retbuf[rbc] = ( isalpha(backup[ic]) ?
+                 tolower(backup[ic]) : backup[ic] );
+            rbc++;
+       }
+       retbuf[rbc] = '\0';
+       /* remove trailing blanks */
+       for(--rbc; isblank(retbuf[rbc]); rbc--) {
+            retbuf[rbc] = '\0';
+       }
+    }
+
     return retbuf;
 }
 
--- cyrus-imapd-2.4.17/lib/imapoptions.authid_normalize	2013-07-05 21:15:20.266577773 +0200
+++ cyrus-imapd-2.4.17/lib/imapoptions	2013-07-05 21:16:16.378056391 +0200
@@ -1452,6 +1452,11 @@ product version in the capabilities */
    interface, otherwise the user is assumed to be in the default
    domain (if set). */
 
+{ "normalizeuid", 0, SWITCH }
+/* Lowercase uid and strip leading and trailing blanks. It is recommended
+   to set this to yes, especially if OpenLDAP is used as authentication
+   source. */
+
 # {"xlist-*", NULL, STRING }
 /* Cyrus 2.4 ONLY - enable special use flags.  The valid names
    can be taken from GMail's XLIST documents or RFC 6154.
--- cyrus-imapd-2.4.17/lib/libcyr_cfg.c.authid_normalize	2012-12-01 20:57:54.000000000 +0100
+++ cyrus-imapd-2.4.17/lib/libcyr_cfg.c	2013-07-05 21:15:20.267577781 +0200
@@ -158,6 +158,10 @@ struct cyrusopt_s cyrus_options[] = {
       CFGVAL(long, 1),
       CYRUS_OPT_SWITCH },
 
+    { CYRUSOPT_NORMALIZEUID,
+      CFGVAL(long, 1),
+      CYRUS_OPT_SWITCH },
+
     { CYRUSOPT_LAST, { NULL }, CYRUS_OPT_NOTOPT }
 };
 
--- cyrus-imapd-2.4.17/lib/libcyr_cfg.h.authid_normalize	2012-12-01 20:57:54.000000000 +0100
+++ cyrus-imapd-2.4.17/lib/libcyr_cfg.h	2013-07-05 21:15:20.267577781 +0200
@@ -116,6 +116,8 @@ enum cyrus_opt {
     CYRUSOPT_SQL_USESSL,
     /* Checkpoint after every recovery (OFF) */
     CYRUSOPT_SKIPLIST_ALWAYS_CHECKPOINT,
+    /* Lowercase uid and strip leading and trailing blanks (OFF) */
+    CYRUSOPT_NORMALIZEUID,
 
     CYRUSOPT_LAST
     
